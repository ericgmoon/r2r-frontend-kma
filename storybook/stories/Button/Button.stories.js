import { action } from "@storybook/addon-actions";
import { text } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react-native";
import React from "react";
import { Text } from "react-native";
import Button from "../../../src/packages/common/button";
import CenterView from "../CenterView";

storiesOf("Button", module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add("with text", () => <Button>I am a button</Button>)
  .add("with some emoji", () => <Button>😀 😎 👍 💯</Button>);
