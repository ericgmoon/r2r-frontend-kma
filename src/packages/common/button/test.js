import React from "react";
import renderer from "react-test-renderer";
import Button from "./index.js";

describe("should capitalise text", () => {
  import React from "react";
  import renderer from "react-test-renderer";
  import Button from "./index.js";

  describe("should capitalise text", () => {
    const node = mount(<Button>button</Button>);
    expect(node.prop("title")).toEqual("BUTTON");
  });

  describe("should trigger onPress", () => {
    const callback = sinon.spy();
    const node = shallow(
      <Button className="common-button" onPress={callback}>
        button
      </Button>
    );
    node.find(".common-button").simulate("click");
    sinon.assert.called(callback);
  });

  describe("renders correctly", () => {
    test("without props", () => {
      const node = renderer.create(<Button>Button</Button>).toJSON();
      expect(node).toMatchSnapshot();
    });

    test("with icon", () => {
      const node = renderer
        .create(<Button icon="east">Button</Button>)
        .toJSON();
      expect(node).toMatchSnapshot();
    });

    test("with icon on left", () => {
      const node = renderer
        .create(
          <Button icon="east" iconLeft>
            Button
          </Button>
        )
        .toJSON();
      expect(node).toMatchSnapshot();
    });

    test("with custom styles", () => {
      const node = renderer
        .create(
          <Button
            icon="east"
            buttonStyle={{ paddingTop: 32, paddingButton: 32 }}
          >
            Button
          </Button>
        )
        .toJSON();
      expect(node).toMatchSnapshot();
    });
  });

  const node = mount(<Button>button</Button>);
  expect(node.prop("title")).toEqual("BUTTON");
});

describe("should trigger onPress", () => {
  const callback = sinon.spy();
  const node = shallow(
    <Button className="common-button" onPress={callback}>
      button
    </Button>
  );
  node.find(".common-button").simulate("click");
  sinon.assert.called(callback);
});

describe("renders correctly", () => {
  test("without props", () => {
    const node = renderer.create(<Button>Button</Button>).toJSON();
    expect(node).toMatchSnapshot();
  });

  test("with icon", () => {
    const node = renderer.create(<Button icon="east">Button</Button>).toJSON();
    expect(node).toMatchSnapshot();
  });

  test("with icon on left", () => {
    const node = renderer
      .create(
        <Button icon="east" iconLeft>
          Button
        </Button>
      )
      .toJSON();
    expect(node).toMatchSnapshot();
  });

  test("with custom styles", () => {
    const node = renderer
      .create(
        <Button icon="east" buttonStyle={{ paddingTop: 32, paddingButton: 32 }}>
          Button
        </Button>
      )
      .toJSON();
    expect(node).toMatchSnapshot();
  });
});
