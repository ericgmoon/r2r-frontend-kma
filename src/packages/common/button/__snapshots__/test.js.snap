// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with custom styles 1`] = `
<View
  style={
    Object {
      "borderRadius": 36,
      "overflow": "hidden",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "background": "#000",
          "backgroundColor": "black",
          "borderColor": "black",
          "borderRadius": 36,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 8,
          "paddingBottom": 16,
          "paddingButton": 32,
          "paddingLeft": 20,
          "paddingRight": 12,
          "paddingTop": 32,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
            "paddingVertical": 1,
            "textAlign": "center",
          }
        }
      >
        BUTTON
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly with icon on left 1`] = `
<View
  style={
    Object {
      "borderRadius": 36,
      "overflow": "hidden",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "background": "#000",
          "backgroundColor": "black",
          "borderColor": "black",
          "borderRadius": 36,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 8,
          "paddingBottom": 16,
          "paddingLeft": 12,
          "paddingRight": 20,
          "paddingTop": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
            "paddingVertical": 1,
            "textAlign": "center",
          }
        }
      >
        BUTTON
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly with icon on right 1`] = `
<View
  style={
    Object {
      "borderRadius": 36,
      "overflow": "hidden",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "background": "#000",
          "backgroundColor": "black",
          "borderColor": "black",
          "borderRadius": 36,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 8,
          "paddingBottom": 16,
          "paddingLeft": 20,
          "paddingRight": 12,
          "paddingTop": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
            "paddingVertical": 1,
            "textAlign": "center",
          }
        }
      >
        BUTTON
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly without props 1`] = `
<View
  style={
    Object {
      "borderRadius": 36,
      "overflow": "hidden",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "background": "#000",
          "backgroundColor": "black",
          "borderColor": "black",
          "borderRadius": 36,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 8,
          "paddingBottom": 16,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
            "paddingVertical": 1,
            "textAlign": "center",
          }
        }
      >
        BUTTON
      </Text>
    </View>
  </View>
</View>
`;
